### PR TITLE
fix GET /k8sd/cluster panic when not initialized

### PR DESCRIPTION
**Pull request by neoaggelos**
### Summary

Prevent panics in `GET /k8sd/cluster` if node is not yet initialized. This is a regression caused as a side-effect of https://github.com/canonical/k8s-snap/pull/257/files#diff-86fdc39de0c5dec676bb358c40f6de2447b0f4c4570315906d9b1c0e6e2b50eaR11-R19

Also perform some more cleanup in the code (do not retrieve the cluster config twice).

### CODE REVIEW

File: src/k8s/pkg/k8sd/api/cluster.go [hunk 1]
```diff
@@ -13,27 +13,21 @@ import (
...
 	if err != nil {
 		return response.InternalError(fmt.Errorf("failed to get cluster members: %w", err))
 	}
-
 	config, err := utils.GetClusterConfig(s.Context, s)
-	if err != nil {
-		return response.InternalError(fmt.Errorf("failed to get user-facing cluster config: %w", err))
-	}
-
-	clusterConfig, err := utils.GetClusterConfig(s.Context, s)
```
Comment (addyess) [inline src/k8s/pkg/k8sd/api/cluster.go:28]: was this just duplicate code?

File: src/k8s/pkg/k8sd/api/cluster.go [hunk 2]
```diff
@@ -13,27 +13,21 @@ import (
 )
 
 func (e *Endpoints) getClusterStatus(s *state.State, r *http.Request) response.Response {
-	snap := e.provider.Snap()
+	// fail if node is not initialized yet
+	if !s.Database.IsOpen() {
+		return response.Unavailable(fmt.Errorf("daemon not yet initialized"))
```
Comment (addyess) [inline src/k8s/pkg/k8sd/api/cluster.go:18]: seems like `Unavailable` is still in the 5xx space (503) so this shouldn't change anything on charm

**Review by addyess — APPROVED** [review]
Nice thank you

File: src/k8s/pkg/k8sd/api/cluster.go (hunk 1 above)
Comment (neoaggelos) [reply src/k8s/pkg/k8sd/api/cluster.go:28]: yes

File: src/k8s/pkg/k8sd/api/cluster.go (hunk 2 above)
Comment (neoaggelos) [reply src/k8s/pkg/k8sd/api/cluster.go:18]: This is trying to imitate https://github.com/canonical/microcluster/blob/5043bc984c83e9ecfdba1de731ec05678b166f13/internal/rest/rest.go#L187-L196 as close as possible

**Pull request merged**